devices/HAA/main.c

```diff
@@ -7483,6 +7484,12 @@ void normal_mode_init() {
         lightbulb_group->armed_autodimmer = false;
         lightbulb_group->autodimmer_task_delay = AUTODIMMER_TASK_DELAY_DEFAULT;
         lightbulb_group->autodimmer_task_step = AUTODIMMER_TASK_STEP_DEFAULT;
+        lightbulb_group->funcs.start = adv_pwm_start;
+        lightbulb_group->funcs.stop = adv_pwm_stop;
+        lightbulb_group->funcs.set_freq = adv_pwm_set_freq;
+        lightbulb_group->funcs.set_duty = adv_pwm_set_duty;
+        lightbulb_group->funcs.get_duty = adv_pwm_get_duty;
+        lightbulb_group->funcs.new_channel = adv_pwm_new_channel;
```

Comment:
Those declarations are not needed. Waste of RAM. LIGHTBULB_TYPE must be used.

Reply:
Do you suggest for each adv_pwm function call check the type of bulb and either call pwn or my92xx driver?